### ISSUE TITLE
Sell quantity lower than exchange min trade amount 

### ISSUE THREAD

**Issue by TimothyClayton** [AlbatrossAutomated/blue_collar_gdax#2]
First encountered recently. This was during very high exchange trading activity. Seems either the exchange returned a `"settled": true` with an inaccurate `"filled_size"`, a bug has been introduced with new code, or the issue arises when BlueCollar runs non-locally.  
 
```  
"Checking BID: 187.58"
2018-03-08T01:33:53.634009+00:00 app[run.2923]: - "Trades flipped: 0"
2018-03-08T01:33:53.634010+00:00 app[run.2923]: - "Checking BID: 187.58"
2018-03-08T01:33:53.634011+00:00 app[run.2923]: - D, [2018-03-08T01:33:52.546750 #4] DEBUG -- :    (1.8ms)  SELECT COUNT(*) FROM "flipped_trades" WHERE "flipped_trades"."sell_pending" = $1  [["sell_pending", "f"]]
2018-03-08T01:33:53.634012+00:00 app[run.2923]: - "Trades flipped: 0"
2018-03-08T01:33:53.634013+00:00 app[run.2923]: - "Checking BID: 187.58"
2018-03-08T01:33:53.634014+00:00 app[run.2923]: - D, [2018-03-08T01:33:53.247630 #4] DEBUG -- :    (1.8ms)  SELECT COUNT(*) FROM "flipped_trades" WHERE "flipped_trades"."sell_pending" = $1  [["sell_pending", "f"]]
2018-03-08T01:33:53.634015+00:00 app[run.2923]: - "Trades flipped: 0"
2018-03-08T01:33:53.634016+00:00 app[run.2923]: - "The SCRUM filled"
2018-03-08T01:33:53.634017+00:00 app[run.2923]: - D, [2018-03-08T01:33:53.630556 #4] DEBUG -- :    (1.5ms)  BEGIN
2018-03-08T01:33:53.634024+00:00 app[run.2923]: - D, [2018-03-08T01:33:53.633757 #4] DEBUG -- :   SQL (2.0ms)  INSERT INTO "flipped_trades" ("trade_pair", "base_currency_purchased", "buy_price", "cost", "buy_order_id", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING "id"  [["trade_pair", "LTC-USD"], ["base_currency_purchased", "0.00013058"], ["buy_price", "187.58"], ["cost", "0.0244941964"], ["buy_order_id", "ddd4589e-53e3-455a-84ef-06506de63a5c"], ["created_at", "2018-03-08 01:33:53.630750"], ["updated_at", "2018-03-08 01:33:53.630750"]]
2018-03-08T01:34:05.171999+00:00 heroku[run.6057]: - Awaiting client
2018-03-08T01:34:05.234461+00:00 heroku[run.6057]: - Starting process with command `bash`
2018-03-08T01:34:05.303233+00:00 heroku[run.6057]: - State changed from starting to up
2018-03-08T01:34:05.974811+00:00 app[run.2923]: - D, [2018-03-08T01:33:53.636504 #4] DEBUG -- :    (2.3ms)  COMMIT
2018-03-08T01:34:05.974840+00:00 app[run.2923]: - "Buy fees incurred: 0.0"
2018-03-08T01:34:05.974842+00:00 app[run.2923]: - "Selling at 187.88 for an estimated profit of 0.00003917 USD and 0.0 LTC."
2018-03-08T01:34:05.974844+00:00 app[run.2923]: - "LIMIT SELL: 0.00013058 LTC @ $187.88"
2018-03-08T01:34:05.974846+00:00 app[run.2923]: - "API Error: {\"message\":\"size is too small. Minimum size is 0.1\"}, Msg: {\"message\":\"size is too small. Minimum size is 0.1\"}, Class: Coinbase::Exchange::BadRequestError"
2018-03-08T01:34:05.974849+00:00 app[run.2923]: - "API Error: {\"message\":\"size is too small. Minimum size is 0.1\"}, Msg: {\"message\":\"size is too small. Minimum size is 0.1\"}, Class: Coinbase::Exchange::BadRequestError"
2018-03-08T01:34:05.974851+00:00 app[run.2923]: - "LIMIT SELL: 0.00013058 LTC @ $187.88"
2018-03-08T01:34:05.974848+00:00 app[run.2923]: - "LIMIT SELL: 0.00013058 LTC @ $187.88"  
```

**Comment by TimothyClayton** [comment]
Replicated. Notice `"filled_size"` in below. New bad response from GDAX (wrong filled size): 
```  
"BUY DOWN FILLED!!"
"Possible API response w/wrong 'filled_size' causing error on sell side: "
{
                "id" => "54b53106-1f57-4b21-8d6a-a208189448ae",
             "price" => "173.83000000",
              "size" => "0.11000000",
        "product_id" => "LTC-USD",
              "side" => "buy",
              "type" => "limit",
     "time_in_force" => "GTC",
         "post_only" => false,
        "created_at" => "2018-03-21T04:59:00.932119Z",
           "done_at" => "2018-03-21T04:59:26.509Z",
       "done_reason" => "filled",
         "fill_fees" => "0.0000000000000000",
       "filled_size" => "0.00011579",
    "executed_value" => "0.0201277757000000",
            "status" => "done",
           "settled" => true
}
```

Here is the response from same endpoint shortly after bad response:  
```  
{
                "id" => "54b53106-1f57-4b21-8d6a-a208189448ae",
             "price" => "173.83000000",
              "size" => "0.11000000",
        "product_id" => "LTC-USD",
              "side" => "buy",
              "type" => "limit",
     "time_in_force" => "GTC",
         "post_only" => false,
        "created_at" => "2018-03-21T04:59:00.932119Z",
           "done_at" => "2018-03-21T04:59:26.509Z",
       "done_reason" => "filled",
         "fill_fees" => "0.0000000000000000",
       "filled_size" => "0.11000000",
    "executed_value" => "19.1213000000000000",
            "status" => "done",
           "settled" => true
}
```

**Comment by TimothyClayton** [comment]
Fix implement at `0a308248bfdb07edbaef9229e793390b0667ec81`
